### PR TITLE
Custom "weights_fpath" load enhancement

### PR DESCRIPTION
**Pull request by yesseyev**
I'm using **Resemblyzer** for a while, but right now I need to use my own **pretrained.pt** file instead of built-in one. 
This enhancement extracts defined in code **weights_fpath**  as constructor argument with default value _None_. If argument is provided, custom model will be used, otherwise Resemblyzer will work as usual.

No major changes were made. Looking forward to see my changes in new version. Thank you in advance.

### CODE REVIEW

File: resemblyzer/voice_encoder.py [hunk 1]
```diff
@@ -9,38 +9,43 @@
 
 
 class VoiceEncoder(nn.Module):
-    def __init__(self, device: Union[str, torch.device]=None, verbose=True):
+    def __init__(self, device: Union[str, torch.device]=None, verbose=True, weights_fpath: Union[Path, str]=None):
```
Comment (CorentinJ) [inline resemblyzer/voice_encoder.py:12]: Put it as first argument (and update references accordingly)

File: resemblyzer/voice_encoder.py [hunk 2]
```diff
@@ -9,38 +9,43 @@
...
         """
-        :param device: either a torch device or the name of a torch device (e.g. "cpu", "cuda"). 
+        :param device: either a torch device or the name of a torch device (e.g. "cpu", "cuda").
+        :param weights_fpath: path to "<CUSTOM_MODEL>.pt" file path
```
Comment (CorentinJ) [inline resemblyzer/voice_encoder.py]: precise that leaving it as none will default to the default model

File: resemblyzer/voice_encoder.py [hunk 3]
```diff
@@ -9,38 +9,43 @@
...
         if device is None:
             device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
         elif isinstance(device, str):
             device = torch.device(device)
         self.device = device
-            
+
         # Load the pretrained model'speaker weights
-        weights_fpath = Path(__file__).resolve().parent.joinpath("pretrained.pt")
+        if not weights_fpath:
```
Comment (CorentinJ) [inline resemblyzer/voice_encoder.py]:
```suggestion
        if weights_fpath is None:
```

File: resemblyzer/voice_encoder.py [hunk 4]
```diff
@@ -9,38 +9,43 @@
...
+            weights_fpath = Path(__file__).resolve().parent.joinpath("pretrained.pt")
+        elif isinstance(weights_fpath, str):
```
Comment (CorentinJ) [inline resemblyzer/voice_encoder.py]:
```suggestion
        else:
```

**Review by CorentinJ — CHANGES_REQUESTED** [review]

File: resemblyzer/voice_encoder.py (hunk 1 above)
Comment (yesseyev) [reply resemblyzer/voice_encoder.py:12]: Wouldn't that produce some back-compatibility issues if some developers will use **device** as _positional argument_, not _keyword argument_?

File: resemblyzer/voice_encoder.py (hunk 2 above)
Comment (yesseyev) [reply resemblyzer/voice_encoder.py]: Sure

File: resemblyzer/voice_encoder.py (hunk 3 above)
Comment (yesseyev) [reply resemblyzer/voice_encoder.py]: Okay

File: resemblyzer/voice_encoder.py (hunk 4 above)
Comment (yesseyev) [reply resemblyzer/voice_encoder.py]: Done

File: resemblyzer/voice_encoder.py (hunk 1 above)
Comment (CorentinJ) [reply resemblyzer/voice_encoder.py:12]: Right I forgot this was Resemblyzer, backwards compat is ideal.

File: resemblyzer/voice_encoder.py (hunk 2 above)
Comment (CorentinJ) [reply resemblyzer/voice_encoder.py]: I don't see your changes, did you push?

**Comment by yesseyev** [comment]
@CorentinJ Resolved all issues.

P.S. My IDEA automatically formatted all lines in before commit hook. Sorry for that.

**Comment by CorentinJ** [comment]
That's fine, this is the code format they should have been in in the first place

**Review by CorentinJ — APPROVED** [review]

**Comment by yesseyev** [comment]
Excuse me @CorentinJ. I wanted to know when would my PR be merged? Thank you!

**Comment by CorentinJ** [comment]
Oh I thought you had merged it

**Comment by Josseca** [comment]
Código

**Pull request merged**